File: Algorithms/Python3.x/451-Sort_Characters_By_Frequency.py
# Runtime: 36 ms, faster than 98.89% of Python3 online submissions for Sort Characters By Frequency.
# Memory Usage: 13.7 MB, less than 78.57% of Python3 online submissions for Sort Characters By Frequency.
from collections import Counter


class Solution:
    def frequencySort(self, s: str) -> str:
        return ''.join([char * times for char, times in Counter(s).most_common()])

File: Algorithms/C++/451-Sort_Characters_By_Frequency.cpp
// Runtime: 40 ms, faster than 16.73% of C++ online submissions for Sort Characters By Frequency.
// Memory Usage: 10.2 MB, less than 100.00% of C++ online submissions for Sort Characters By Frequency.
class Solution {
public:
    string frequencySort(string s) {
        unsigned int counts[256] = {0};
        for (char ch : s)
            ++counts[ch];
        sort(s.begin(), s.end(), [&] (char a, char b) {
            return counts[a] > counts[b] || (counts[a] == counts[b] && a < b);
        });
        return s;
    }
};
